app/forms/steps/client/contact_details_form.rb

```diff
@@ -24,6 +24,7 @@ def telephone_number=(str)
       def choices
         values = CorrespondenceType.values.dup
         values.delete(CorrespondenceType::HOME_ADDRESS) unless applicant.home_address?
+        values.delete(CorrespondenceType::HOME_ADDRESS) if no_residence_type?
```

Comment:
No harm leaving this, but I can't get my brain to calculate when this would get hit. The guard clause above I think covers us (along with the resetting of home address you're doing in the residence type form) 

Not super bothered if you see an edge case/just want the peace of mind